snr.py

```diff
@@ -0,0 +1,287 @@
+import numpy as np
+import astropy.units as u
+from astropy.tests.helper import assert_quantity_allclose
+from scipy.optimize import fsolve
```

Comment:
@pllim: Does synphot require scipy, or would we be introducing a new dependency?

Reply:
It uses scipy because Tabular1D does. No worries there.